infra/lib/lambda/lambda-stack.ts

```diff
@@ -127,6 +138,36 @@ export class LambdaStack extends cdk.Stack {
       },
     });
 
+    this.handleTopicMessageFn = new lambda.Function(this, "handleTopicMessageFn", {
+      runtime: lambda.Runtime.NODEJS_20_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(path.join(__dirname, "./src/handle-topic-message")),
+    });
+
+    this.uploadMovieFn.addToRolePolicy(new iam.PolicyStatement({
+      actions: [
+        'sqs:SendMessage',
+      ],
+      resources: ['*'], // Adjust as needed for more specific permissions
+    }));
+
+    this.subscribeFn.addToRolePolicy(new iam.PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: [
+            'cognito-idp:AdminGetUser',
+            'sqs:SendMessage',
+        ],
+        resources: ['*'],
+    }));
+
+    this.handleTopicMessageFn.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['sns:*'],
+      resources: ['*'],
+    }));
+
+    this.handleTopicMessageFn.addEventSource(new SqsEventSource(sqsQueue));
```

Comment:
I guess there are statements to allow for this instead of writing policies yourself.

Reply:
Not for admin get user though.